Stanford
 
## Abstract 

Identifying transcriptional enhancers and their target genes is essential for understanding gene regulation and the impact of human genetic variation on disease. 

We create and evaluate a resource of >13 million enhancer-gene regulatory interactions across 352 cell types and tissues, by integrating predictive models, measurements of chromatin state and 3D contacts, and large-scale genetic perturbations generated by the ENCODE Consortium.

**To Check:** CRISPR perturbation experiments, >30,000 fine-mapped eQTLs, and 569 fine-mapped GWAS variants linked to a likely causal gene.

Create ENCODE-rE2G, we build an encyclopedia of enhancer-gene regulatory interactions in the human genome, which reveals global properties of enhancer networks, identifies differences in the functions of genes that have more or less complex regulatory landscapes, and improves analyses to link noncoding variants to target genes and cell types for common, complex diseases.


## Introduction 
Noncoding DNA elements called enhancers have essential functions in controlling cell-type specific gene regulation and cellular programs, and likely contain a majority of causal variants for common, complex diseases[4](https://www.biorxiv.org/content/10.1101/2023.11.09.563812v1.full#ref-4)–[6](https://www.biorxiv.org/content/10.1101/2023.11.09.563812v1.full#ref-6). It remains an open challenge to accurately identify which elements act as enhancers in a given cell type and link them to the nearby genes that they regulate (hereafter, “enhancer-gene regulatory interactions”.

The ENCODE Project has conducted thousands of experiments to identify candidate _cis_-regulatory elements and annotate their chromatin state and 3D physical interactions across hundreds of cell types and tissues[7](https://www.biorxiv.org/content/10.1101/2023.11.09.563812v1.full#ref-7)


Previous predictive models have identified certain features important for prediction accuracy, including the [[strength of activating chromatin marks at an element (“enhancer activity”)]] and the [[frequency at which an element physically contacts a nearby promoter (“3D contact”)]][9](https://www.biorxiv.org/content/10.1101/2023.11.09.563812v1.full#ref-9)

Data collected in this final phase of the ENCODE Project include CRISPR perturbations of candidate enhancers[9](https://www.biorxiv.org/content/10.1101/2023.11.09.563812v1.full#ref-9),[18](https://www.biorxiv.org/content/10.1101/2023.11.09.563812v1.full#ref-18)–[20](https://www.biorxiv.org/content/10.1101/2023.11.09.563812v1.full#ref-20), high-resolution Hi-C data[21](https://www.biorxiv.org/content/10.1101/2023.11.09.563812v1.full#ref-21), and maps of DNase I hypersensitivity across new cell types[22](https://www.biorxiv.org/content/10.1101/2023.11.09.563812v1.full#ref-22) — providing an opportunity to build a common benchmarking framework, build improved predictive models, and apply these models to construct a map of enhancer-gene regulatory interactions in the human genome.


## Figure 1

### a
#### 1
A new set of predictive models, collectively termed “ENCODE-rE2G”**. We developed new supervised classifiers that integrate molecular features of chromatin state and 3D physical interactions to predict enhancer-gene regulatory interactions in a given cell type. ENCODE-rE2G involves supervised training directly on CRISPR perturbation data, and enables predicting which enhancers regulate which genes in a given cell type based on various combinations of cell-type specific experimental data (minimally, DNase-seq combined with a reference ENCODE Hi-C map averaged across many tissues).

#### 2
**A new benchmarking framework to evaluate predictive models.** We collected and harmonized genetic perturbation data relevant to enhancer-gene regulatory interactions, including CRISPR perturbation experiments, expression quantitative trait loci (eQTLs), and variants from genome-wide association studies (GWAS). We developed a pipeline to quantify and annotate the accuracy of predictive models, and applied this to benchmark ENCODE-rE2G and hundreds of alternative models at key prediction tasks. An important difference between this and some previous benchmarking analyses is that we use as gold standards only data describing _regulatory interactions_, derived from genetic perturbations, as opposed to _physical interactions_, such as those measured by Hi-C or ChIA-PET. These perturbations datasets, analysis pipelines, and benchmarking results will facilitate studies to develop improved models of enhancer-gene regulatory interactions.

#### 3
**An encyclopedia of enhancer-gene regulatory interactions in the human genome**. We applied the most general of the ENCODE-rE2G models across 352 ENCODE cell types, tissues, and contexts (collectively, “biosamples”), and identified 13,455,443 regulatory interactions between distal elements and target genes. This atlas includes an average of 23,846 unique regulatory elements and 62,509 element-gene regulatory interactions per biosample, and in total annotates 0.76% of the approximately 3 billion basepairs in the human genome. This resource enables looking up a gene to find predicted enhancers, looking up an element to find predicted target genes, and identifying predicted cell types and target genes for noncoding variants.

#nih #ncbi #bio #paper #good4pub 
