In 1988, Richard Lenski started growing a common strain of _E. coli_ bacteria in twelve identical Petri dishes in his laboratory at Michigan State University. Since then they have reproduced about 7 times per day, adding up to over 70 thousand generations. Lenski's team has periodically collected samples from each dish for full genome sequencing.

This 30-year experiment has allowed the rates of mutation and the rate of adaptation to be directly measured in a real-world system.

Looking at sequences collected from an individual bacterium in the Lenski culture is like capturing snapshots of evolution.
In a given gene of the _E. coli_ genome, single nucleotide polymorphism (SNP) mutations occur rarely. Sometimes these mutations persist in the population and turn up in individual samples taken hundreds of generations later, but most of the time the mutation is transient, disappearing just as rapidly as it appeared.
![[Pasted image 20240305143716.png]]
A SNP which increases the reproductive fitness of an organism is called **adaptive** and, with an increased reproductive rate, tends to take over the entire population — a process called **fixation**.

A **deleterious** SNP decreases the reproductive fitness of an organism and usually leads to an evolutionary dead end which quickly disappears from the gene pool.

A **synonymous** SNP is one that does not change a translated protein sequence and thus has no effect on reproductive fitness. The red mutations are likely deleterious or synonymous.

However, synonymous SNPs which occur in a single organism can sometimes become fixed in a population with a very small probability 1/N, where N is the population (billions in the case of bacterial cultures).

This process is called **genetic drift**.

![[Pasted image 20240305144226.png]]