Regulatory elements, such as enhancers, play a critical role in orchestrating gene expression, ensuring precise spatial and temporal control necessary for development, homeostasis, and adaptation. Variants in these non-coding regions have emerged as key contributors to phenotypic diversity and disease susceptibility. Genome-wide association studies (GWAS) have revolutionized our understanding of genetic contributions to complex traits, revealing that over 90% of disease-associated variants lie within non-coding regions, particularly enhancers. These findings underscore the importance of regulatory variation in modulating gene expression and linking genetic changes to organismal phenotypes

Genome-wide association studies (GWAS) have revealed that the majority of disease-associated genetic variants occur in non-coding regions of the genome, particularly within regulatory elements that modulate gene expression~\cite{uffelmann2021gwas, visscher2017ten, knight2014approaches}.Â These regulatory variants can have profound effects on phenotypes and disease susceptibility by altering the complex mechanisms that control when, where, and how much genes are expressed~\cite{albert2015regulatory}. 


These models not only prioritize candidate causal variants but also reveal the molecular mechanisms by which these variants influence gene regulation. 


By integrating statistical fine-mapping with deep learning predictions, researchers have identified variants that likely contribute to complex traits and diseases, underscoring the utility of these approaches in understanding the genetic basis of human phenotypes


Selecting the most suitable model for detecting the regulatory effects of genetic variants remains a significant challenge, despite several surveys offering detailed overviews of the deep learning ecosystem in this domain. These surveys have highlighted the unique strengths and limitations of various models. However, they lack a unified framework for assessment. Specifically, existing reviews often fail to benchmark models on a standardized dataset, train or fine-tune them under consistent conditions, and evaluate their performance using uniform criteria. This inconsistency hinders objective evaluation and comparability across different approaches. Addressing this gap through systematic benchmarking is crucial to providing fair performance assessments and guiding the selection of models for diverse applications. 

In this study, we evaluated state-of-the-art deep learning models for predicting the effects of genetic variants on enhancer activity in the human genome. To achieve this, we collected and curated nine datasets derived from diverse experimental approaches, including massively parallel reporter assays (MPRA), reporter assay quantitative trait loci (raQTL), and expression quantitative trait loci (eQTL) studies. These datasets provided a comprehensive benchmark by covering 54,859 single-nucleotide polymorphisms (SNPs) located in enhancer regions across four distinct human cell lines.


In this study, we evaluate state-of-the-art deep learning models to predict the effects of genetic variants on enhancer activity in the human genome. Our approach involved curating and integrating nine datasets derived from diverse experimental methodologies, including massively parallel reporter assays (MPRA), reporter assay quantitative trait loci (raQTL), and expression quantitative trait loci (eQTL) studies. These datasets encompassed 54,859 single-nucleotide polymorphisms (SNPs) located in enhancer regions across four distinct human cell lines. To ensure a robust and thorough assessment, we fine-tuned 22 deep learning models, systematically exploring a broad spectrum of architectural designs and parameter configurations. 