

Noncoding genomic variations constitute the majority of disease and other trait-associated [[single-nucleotide polymorphisms]] (SNPs) but characterizing their functional effects remains a challenge. 


Recent progress on prioritizing functional noncoding variants has been made by integrating evolutionary conservation and genomic and [[chromatin annotations]] at the position of interest. 

Such approaches are valuable for prioritizing sequence variants; however, current methods except for a parallel work have not been able to extract and utilize regulatory sequence information de novo for noncoding-variant function prediction, which requires precise allele-specific prediction with single-nucleotide sensitivity. In fact, no previous approach predicts functional effects of noncoding variants from only genomic sequence, and no method has been demonstrated to predict with single-nucleotide sensitivity the effects of noncoding variants on [[transcription factor (TF) binding]], [[DNA accessibility]] and [[histone marks of sequences]].