## Main Questions 

### Question 1

- Detection of the motif of the TF matching the TF from ChIP-seq experiments. If you are unable to identify the expected motif, what are the reasons? 

- Would other tools (Borzoit, etc) be able to detect the motif? 

- How close is the motif located to the center of TF ChIP-seq peaks? DONE

- What is the percentage of ChIP-seq peaks with the detected motif? Comparing to the classical footprinting methods (ATAC-seq like), what is the resolution of your method?

### Question 2

- Detection of co-factors. How does your method compares to MuMod and similar approaches? 
- Does the presence of co-factors 'shift' the main motif off-center (this would be indicative of ChIP-seq targeting the complex, in which the original TF participates as opposed to the independent binding of that TF)? 
- Do some TFs tend to form complexes while others rarely do so? 
- What do you get in case of overbinders (Sanjar's study)—for over 50% of ChIP-seq peaks, you will find overlapping with dozens of other ChIP-seq experiments making the peaks non-specific?